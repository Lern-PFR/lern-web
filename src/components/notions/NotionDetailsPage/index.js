import LessonContent from './LessonContent';
import Sidebar from './Sidebar';

export {
	LessonContent,
	Sidebar,
};
