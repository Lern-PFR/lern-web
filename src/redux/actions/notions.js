import * as NotionsApi from 'api/notionsApi';

/**
 * @constant
 * @name ActionTypes
 * @description The various action types used to interact with the notions redux state.
 * @type {object}
 *
 * @author Timothée Simon-Franza
 */
export const ActionTypes = {
	FETCH_NOTION_REQUEST: '@NOTIONS/FETCH_REQUEST',
	FETCH_NOTION_SUCCESS: '@NOTIONS/FETCH_SUCCESS',
	FETCH_NOTION_FAILURE: '@NOTIONS/FETCH_FAILURE',

	FETCH_NOTION_LIST_REQUEST: '@NOTIONS/FETCH_LIST_REQUEST',
	FETCH_NOTION_LIST_SUCCESS: '@NOTIONS/FETCH_LIST_SUCCESS',
	FETCH_NOTION_LIST_FAILURE: '@NOTIONS/FETCH_LIST_FAILURE',

	CREATE_NOTION_REQUEST: '@NOTIONS/CREATE_REQUEST',
	CREATE_NOTION_SUCCESS: '@NOTIONS/CREATE_SUCCESS',
	CREATE_NOTION_FAILURE: '@NOTIONS/CREATE_FAILURE',

	UPDATE_NOTION_REQUEST: '@NOTIONS/UPDATE_REQUEST',
	UPDATE_NOTION_SUCCESS: '@NOTIONS/UPDATE_SUCCESS',
	UPDATE_NOTION_FAILURE: '@NOTIONS/UPDATE_FAILURE',

	DELETE_NOTION_REQUEST: '@NOTIONS/DELETE_REQUEST',
	DELETE_NOTION_SUCCESS: '@NOTIONS/DELETE_SUCCESS',
	DELETE_NOTION_FAILURE: '@NOTIONS/DELETE_FAILURE',

	CLEAR_NOTION_LIST: '@NOTIONS/CLEAR_LIST',
};

// //////////////////////////////////////////////////////// //
// //////////// Specific notion fetching actions ////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name fetchNotionRequest
 * @description Action triggered anytime a notion fetching call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const fetchNotionRequest = () => ({ type: ActionTypes.FETCH_NOTION_REQUEST });

/**
 * @function
 * @name fetchNotionSuccess
 * @description Action triggered as a result to a successful notion fetching API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} notion : The notion retrieved from the API.
 *
 * @returns {object}
 */
const fetchNotionSuccess = ({ notion }) => ({
	type: ActionTypes.FETCH_NOTION_SUCCESS,
	payload: { notion },
});

/**
 * @function
 * @name fetchNotionFailure
 * @description Action triggered as a result to a failed notion fetching API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const fetchNotionFailure = (error) => ({
	type: ActionTypes.FETCH_NOTION_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// ////////////// Notion list fetching actions //////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name fetchNotionListRequest
 * @description Action triggered anytime a notion list fetching call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const fetchNotionListRequest = () => ({ type: ActionTypes.FETCH_NOTION_LIST_REQUEST });

/**
 * @function
 * @name fetchNotionListSuccess
 * @description Action triggered as a result to a successful notion list fetching API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {array}	notions		: The list of notions retrieved from the API.
 * @param {number}	totalCount	: The total amount of notions available for the module.
 *
 * @returns {object}
 */
const fetchNotionListSuccess = ({ notions, totalCount }) => ({
	type: ActionTypes.FETCH_NOTION_LIST_SUCCESS,
	payload: { notions, totalCount },
});

/**
 * @function
 * @name fetchNotionListFailure
 * @description Action triggered as a result to a failed notion list fetching API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const fetchNotionListFailure = (error) => ({
	type: ActionTypes.FETCH_NOTION_LIST_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// //////////////// Notion creation actions /////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name createNotionRequest
 * @description Action triggered anytime a notion creation call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const createNotionRequest = () => ({ type: ActionTypes.CREATE_NOTION_REQUEST });

/**
 * @function
 * @name createNotionSuccess
 * @description Action triggered as a result to a successfult notion creation API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} notion : The newly created notion.
 *
 * @returns {object}
 */
const createNotionSuccess = ({ notion }) => ({
	type: ActionTypes.CREATE_NOTION_SUCCESS,
	payload: { notion },
});

/**
 * @function
 * @name createNotionFailure
 * @description Action triggered as a result to a failed notion creation API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const createNotionFailure = (error) => ({
	type: ActionTypes.CREATE_NOTION_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// //////////////// Exported action creators ////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name fetchNotion
 * @description Method used to fetch the latest version of a specified notion.
 *
 * @author Timothée Simon-Franza
 *
 * @param {string} notionId : The id of the notion we want to retrieve.
 */
export const fetchNotion = (notionId) => (dispatch) => {
	dispatch(fetchNotionRequest());

	return NotionsApi.fetchNotionById(notionId)
		.then(({ notion }) => dispatch(fetchNotionSuccess({ notion })))
		.catch((error) => dispatch(fetchNotionFailure(error)));
};

/**
 * @function
 * @name fetchNotionListByModuleId
 * @description Method used to fetch all notions from a specific module, identified by the moduleId param.
 *
 * @author Timothée Simon-Franza
 *
 * @param {string} moduleId : The id of the module we want to retrieve notions from.
 */
export const fetchNotionListByModuleId = (moduleId) => (dispatch) => {
	dispatch(fetchNotionListRequest());

	return NotionsApi.fetchNotionsByModuleId(moduleId)
		.then(({ notions, totalCount }) => dispatch(fetchNotionListSuccess({ notions, totalCount })))
		.catch((error) => dispatch(fetchNotionListFailure(error)));
};

/**
 * @function
 * @name createNotion
 * @description Method used to create a new notion in the database.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} notionData : The data to create the new notion from.
 */
export const createNotion = (notionData) => (dispatch) => {
	dispatch(createNotionRequest());

	return NotionsApi.createNotion(notionData)
		.then(({ notion }) => dispatch(createNotionSuccess({ notion })))
		.catch((error) => dispatch(createNotionFailure(error)));
};
