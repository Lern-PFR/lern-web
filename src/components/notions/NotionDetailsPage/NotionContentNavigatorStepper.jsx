import PropTypes from 'prop-types';
import styled from 'styled-components';
import { stepper, stepperCurrent } from 'theme/pages/notions/notionDetailsPage';

const StepperButton = styled('button')(
	{
		...stepper,
		backgroundColor: (({ isCurrent }) => (isCurrent ? stepperCurrent.backgroundColor : stepper.backgroundColor)),
	}
);

/**
 * @name NotionContentNavigatorStepper
 * @description A stepper button used to navigate between a notion's content list.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {bool}	isCurrent	Whether the stepper represents the currently displayed document.
 * @param {string}	label		The label to display on hover.
 * @param {func}	onClick		Method to trigger when the user clicks the button.
 */
const NotionContentNavigatorStepper = ({ isCurrent, label, onClick, ...otherProps }) => (
	<StepperButton onClick={onClick} label={label} isCurrent={isCurrent} {...otherProps} />
);

NotionContentNavigatorStepper.propTypes = {
	isCurrent: PropTypes.bool,
	label: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
};

NotionContentNavigatorStepper.defaultProps = {
	isCurrent: false,
};

export default NotionContentNavigatorStepper;
