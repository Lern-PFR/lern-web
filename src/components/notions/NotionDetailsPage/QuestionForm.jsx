import PropTypes from 'prop-types';
import { BodyCopy } from 'components/shared/typography';

import { StyledList, StyledListItem } from 'components/shared/layout';
import { answersList, answerListItem, answerFormSubmitButton } from 'theme/pages/notions/notionDetailsPage';
import { PrimaryButton } from 'components/shared/buttons';
import { withTranslation } from 'react-i18next';

// @TODO: Implement a way to know if the form has been submitted.
// @TODO: Once the form has been submitted, disable the submit button.
// @TODO: Implement the "see explanation" button and its logic.

/**
 * @name QuestionForm
 * @description The form used by the user to answer a lesson's question if there is one.
 *
 * @author Timothée Simon-Franza
 *
 * @param {array}	answers			The list of possible answers to the current questions.
 * @param {func}	onSubmit		The method to trigger upon submission of the form.
 * @param {bool}	[singleChoice]	Whether the question has multiple valid answers.
 */
const QuestionForm = ({ answers, onSubmit, singleChoice, t }) => {
	/**
	 * @name handleSubmit
	 * @description An intermedierary method to handle form submission side effects and format data.
	 *
	 * @author Timothée Simon-Franza
	 *
	 * @param {object} event The form submission event object.
	 */
	const handleSubmit = (event) => {
		event.preventDefault();
		onSubmit(event);
	};

	return (
		<form data-testid="question-form" onSubmit={handleSubmit}>
			<StyledList {...answersList}>
				{answers.map(({ id, isValid, text }) => (
					<StyledListItem key={id} {...answerListItem}>
						<input id={id} type={singleChoice ? 'radio' : 'checkbox'} name="answer" value={id} isValid={isValid} />
						<BodyCopy tag="label" htmlFor={id}>{text}</BodyCopy>
					</StyledListItem>
				))}
			</StyledList>
			<PrimaryButton type="submit" {...answerFormSubmitButton}>{t('exercises.validate-answer')}</PrimaryButton>
		</form>
	);
};

QuestionForm.propTypes = {
	answers: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
		text: PropTypes.string.isRequired,
		valid: PropTypes.bool.isRequired,
	})).isRequired,
	onSubmit: PropTypes.func.isRequired,
	singleChoice: PropTypes.bool,
	t: PropTypes.func.isRequired,
};

QuestionForm.defaultProps = {
	singleChoice: false,
};

export default withTranslation()(QuestionForm);
