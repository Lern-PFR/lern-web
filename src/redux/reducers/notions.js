import { ActionTypes } from 'redux/actions/notions';

const initialState = {
	isLoading: false,
	items: [],
	totalCount: 0,
};

export default (state = initialState, { type, payload }) => {
	switch (type) {
		case ActionTypes.FETCH_NOTION_REQUEST:
		case ActionTypes.FETCH_NOTION_LIST_REQUEST:
		case ActionTypes.CREATE_NOTION_REQUEST:
		case ActionTypes.UPDATE_NOTION_REQUEST:
		case ActionTypes.DELETE_NOTION_REQUEST:
			return {
				...state,
				isLoading: true,
			};
		case ActionTypes.FETCH_NOTION_FAILURE:
		case ActionTypes.FETCH_NOTION_LIST_FAILURE:
		case ActionTypes.CREATE_NOTION_FAILURE:
		case ActionTypes.UPDATE_NOTION_FAILURE:
		case ActionTypes.DELETE_NOTION_FAILURE:
			return {
				...state,
				isLoading: false,
			};
		case ActionTypes.CREATE_NOTION_SUCCESS:
		case ActionTypes.UPDATE_NOTION_SUCCESS:
		case ActionTypes.DELETE_NOTION_SUCCESS:
			return {
				...state,
				isLoading: false,
			};
		case ActionTypes.FETCH_NOTION_SUCCESS:
			return {
				...state,
				isLoading: false,
				items: [payload.notion],
				totalCount: 1,
			};
		case ActionTypes.FETCH_NOTION_LIST_SUCCESS:
			return {
				...state,
				isLoading: false,
				items: [...payload.notions],
				totalCount: payload.totalCount,
			};
		case ActionTypes.CLEAR_NOTION_LIST:
			return {
				...state,
				items: [],
				totalCount: 0,
			};
		default:
			return state;
	}
};
