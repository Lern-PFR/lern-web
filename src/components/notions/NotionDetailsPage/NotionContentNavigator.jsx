import PropTypes from 'prop-types';
import _ from 'lodash';
import { ChevronLeft, ChevronRight } from 'react-feather';
import styled from 'styled-components';
import { StyledDiv, StyledList } from 'components/shared/layout';
import { navigationChevrons, navigator, stepperList } from 'theme/pages/notions/notionDetailsPage';

import NotionContentNavigatorStepper from './NotionContentNavigatorStepper';

// @TODO: Display the name of the document in a tooltip upon hovering a stepper.

const PreviousButton = styled(ChevronLeft)({ ...navigationChevrons });
const NextButton = styled(ChevronRight)({ ...navigationChevrons });

/**
 * @name NotionContentNavigator
 * @description A component used to navigate inside a lesson's pages.
 *
 * @author Timothée Simon-Franza
 *
 * @param {number}	currentDocOrder	The current document's order value.
 * @param {array}	notionContent	An array of the current notion's lessons and exercises.
 * @param {func}	redirectTo		Redirection method to trigger when a stepper is clicked.
 */
const NotionContentNavigator = ({ currentDocOrder, notionContent, redirectTo }) => {
	/**
	 * @name onPreviousClick
	 * @description onClick handler method for the 'previous' icon button.
	 *
	 * @author Timothée Simon-Franza
	 */
	const onPreviousClick = () => {
		if (currentDocOrder > 0) {
			redirectTo(currentDocOrder - 1);
		}
	};

	/**
	 * @name onNextClick
	 * @description onClick handler method for the 'next' icon button.
	 *
	 * @author Timothée Simon-Franza
	 */
	const onNextClick = () => {
		if (currentDocOrder < notionContent.length - 1) {
			redirectTo(currentDocOrder + 1);
		}
	};

	return (
		<StyledDiv {...navigator}>
			<PreviousButton role="button" data-testid="notion-navigatior-previous" onClick={onPreviousClick} disabled={currentDocOrder === 0} background="white" />
			<StyledList {...stepperList}>
				{notionContent.map(({ id, order, title = '', name = '' }) => (
					<NotionContentNavigatorStepper
						isCurrent={_.isEqual(notionContent[currentDocOrder]?.id, id)}
						key={id}
						label={title || name}
						onClick={() => redirectTo(order)}
					/>
				))}
			</StyledList>
			<NextButton role="button" data-testid="notion-navigatior-next" onClick={onNextClick} disabled={currentDocOrder === (notionContent.length - 1)} />
		</StyledDiv>
	);
};

NotionContentNavigator.propTypes = {
	currentDocOrder: PropTypes.number.isRequired,
	notionContent: PropTypes.arrayOf(
		PropTypes.oneOfType([
			PropTypes.shape({
				id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
				title: PropTypes.string,
				name: PropTypes.string,
				order: PropTypes.number.isRequired,
			}),
			PropTypes.shape({
				id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
				name: PropTypes.string.isRequired,
				order: PropTypes.number.isRequired,
			}),
		])
	).isRequired,
	redirectTo: PropTypes.func.isRequired,
};

export default NotionContentNavigator;
