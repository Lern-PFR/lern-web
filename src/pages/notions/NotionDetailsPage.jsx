import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { generatePath } from 'react-router';
import { withTranslation } from 'react-i18next';
import _ from 'lodash';

import routes from 'routes/keys';
import conf from 'conf';
import { notionDetailsMock } from 'mockedData';

import { fetchNotion } from 'redux/actions/notions';
import { StyledDiv, StyledSvg } from 'components/shared/styledElements';
import { LessonContent, Sidebar } from 'components/notions/NotionDetailsPage';
import { SubtleLinkButton } from 'components/shared/buttons';
import { Link } from 'components/shared/navigation';
import { backToModuleSvg, pageLayout } from 'theme/pages/notions/notionDetailsPage';
import { backToParentButtonContentLayout } from 'theme/buttonStyles';
/**
 * @name NotionDetailsPage
 * @description A page used to display a the current notion and its composing lessons.
 *
 * @author Timothée Simon-Franza
 *
 * @param {array}			content				An array of the current notion's lessons and exercises.
 * @param {func}			dispatchFetchNotion	Dispatched action creator used to retrieve the current notion.
 * @param {number|string}	[lessonId]			The current lesson's id.
 * @param {object}			module				The notion's parent module.
 * @param {number|string}	notionId			The current notion's id.
 * @param {object}			notion				The current notion.
 * @param {func}			t					The translation method provided by the withTranslation HoC.
 */
const NotionDetailsPage = ({ content, dispatchFetchNotion, lessonId, module, notionId, notion, t }) => {
	useEffect(() => {
		dispatchFetchNotion(notionId);
	}, [dispatchFetchNotion, notionId]);

	/**
	 * @field currentIndex
	 * @description The index of the current document to display.
	 *
	 * This value should match the order attribute of the lessons and exercises of the current notion.
	 */
	const [currentIndex, setCurrentIndex] = useState(0);

	useEffect(() => {
		if (lessonId) {
			setCurrentIndex(content.filter(({ id }) => _.isEqual(id, lessonId))[0]?.order || 0);
		}
	}, [content, lessonId]);

	/**
	 * @name onQuestionAnswerSubmit
	 * @description The method to trigger upon submission of the question form.
	 *
	 * @author Timothée Simon-Franza
	 */
	const onQuestionAnswerSubmit = () => {
	};

	return (
		<StyledDiv>
			<StyledDiv {...pageLayout}>
				<StyledDiv paddingTop="24px">
					<SubtleLinkButton padding="0">
						<Link to={generatePath(routes.modules.moduleDetails, { moduleId: module.id })}>
							<StyledDiv {...backToParentButtonContentLayout}>
								<StyledSvg src={`${conf.svgPath}/chevronLeft.svg`} {...backToModuleSvg} />
								{t('notions.links.back_to_module', { module })}
							</StyledDiv>
						</Link>
					</SubtleLinkButton>
					<LessonContent
						{...content[currentIndex]}
						question={content[currentIndex].exercise?.question?.statement}
					/>
				</StyledDiv>
				<Sidebar
					currentLesson={content[currentIndex]}
					notionContent={content}
					notionName={notion.title}
					onQuestionAnswerSubmit={onQuestionAnswerSubmit}
					onCurrentDocumentRedirect={(index) => setCurrentIndex(index)}
				/>
			</StyledDiv>
		</StyledDiv>
	);
};

NotionDetailsPage.propTypes = {
	lessonId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	notionId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
	notion: PropTypes.shape({
		id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
		title: PropTypes.string.isRequired,
		description: PropTypes.string.isRequired,
		moduleId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
	}).isRequired,
	content: PropTypes.arrayOf(
		PropTypes.oneOfType([
			// Lessons
			PropTypes.shape({
				id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
				name: PropTypes.string.isRequired,
				description: PropTypes.string.isRequired,
				content: PropTypes.string.isRequired,
				order: PropTypes.number.isRequired,
				contentType: PropTypes.oneOf(['lesson']).isRequired,
				exercise: PropTypes.shape({
					id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
					question: PropTypes.shape({
						id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
						singleChoice: PropTypes.bool.isRequired,
						statement: PropTypes.string.isRequired,
						explanation: PropTypes.string,
						answers: PropTypes.arrayOf(PropTypes.shape({
							text: PropTypes.string.isRequired,
							valid: PropTypes.bool.isRequired,
						})).isRequired,
					}).isRequired,
				}),
			}),
			// Exercises
			PropTypes.shape({
				id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
				order: PropTypes.number.isRequired,
				contentType: PropTypes.oneOf(['exercise']).isRequired,
				question: PropTypes.shape({
					id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
					singleChoice: PropTypes.bool.isRequired,
					statement: PropTypes.string.isRequired,
					explanation: PropTypes.string,
					answers: PropTypes.arrayOf(PropTypes.shape({
						text: PropTypes.string.isRequired,
						valid: PropTypes.bool.isRequired,
					})).isRequired,
				}).isRequired,
			}),
		])
	).isRequired,
	module: PropTypes.shape({
		id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
		name: PropTypes.string.isRequired,
	}).isRequired,
	t: PropTypes.func.isRequired,
	// Dispatched redux action creators
	dispatchFetchNotion: PropTypes.func.isRequired,
};

NotionDetailsPage.defaultProps = {
	lessonId: undefined,
};

/**
 *
 * @param {*} state
 */
const mapStateToProps = (state, ownProps) => {
	const { match: { params: { notionId, lessonId = undefined } } } = ownProps;

	// @TODO: replace with real information once data can be retrieved from the API.
	const { exercises, lessons, modules, notion } = notionDetailsMock;

	const content = [
		...lessons.map((lesson) => ({ ...lesson, contentType: 'lesson' })),
		...exercises.map((exercise) => ({ ...exercise, contentType: 'exercise' })),
	];

	_.sortBy(content, 'order');

	return {
		notionId,
		notion,
		content,
		lessonId,
		module: _.head(_.filter(modules, ({ id }) => _.isEqual(id, notion.moduleId))),
	};
};

const mapDispatchToProps = {
	dispatchFetchNotion: fetchNotion,
};

export default compose(
	withTranslation(),
	connect(mapStateToProps, mapDispatchToProps)
)(NotionDetailsPage);
