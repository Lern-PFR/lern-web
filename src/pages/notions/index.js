import NotionDetailsPage from './NotionDetailsPage';

export {
	// eslint-disable-next-line import/prefer-default-export
	NotionDetailsPage,
};
