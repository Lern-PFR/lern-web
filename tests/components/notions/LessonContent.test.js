import { shallow } from 'enzyme';
import { LessonContent } from 'components/notions/NotionDetailsPage';

describe('LessonContent', () => {
	const lessonData = {
		name: 'test lesson',
		content: 'test lesson content',
		description: 'test lesson description',
	};

	it('should match previous snapshot (no question provided)', () => {
		const wrapper = shallow(<LessonContent {...lessonData} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot (with question provided)', () => {
		const wrapper = shallow(<LessonContent {...lessonData} question="test lesson question" />);
		expect(wrapper).toMatchSnapshot();
	});
});
