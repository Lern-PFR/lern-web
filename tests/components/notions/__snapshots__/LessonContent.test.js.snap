// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonContent should match previous snapshot (no question provided) 1`] = `
<styled.div
  display="grid"
  gridGap="24px"
  gridTemplateRows="7rem 1fr"
  paddingLeft="64px"
>
  <styled.div
    bg="#FFFFFF"
    paddingTop="24px"
    position="sticky"
    top={0}
  >
    <Trafalgar
      lineHeight="initial"
      margin={0}
      tag="h2"
      textTransform="capitalize"
    >
      test lesson
    </Trafalgar>
    <LongPrimer
      margin={0}
    >
      test lesson description
    </LongPrimer>
  </styled.div>
  <styled.div>
    <BodyCopy
      margin={0}
    >
      test lesson content
    </BodyCopy>
  </styled.div>
</styled.div>
`;

exports[`LessonContent should match previous snapshot (with question provided) 1`] = `
<styled.div
  display="grid"
  gridGap="24px"
  gridTemplateRows="7rem 1fr"
  paddingLeft="64px"
>
  <styled.div
    bg="#FFFFFF"
    paddingTop="24px"
    position="sticky"
    top={0}
  >
    <Trafalgar
      lineHeight="initial"
      margin={0}
      tag="h2"
      textTransform="capitalize"
    >
      test lesson
    </Trafalgar>
    <LongPrimer
      margin={0}
    >
      test lesson description
    </LongPrimer>
  </styled.div>
  <styled.div>
    <BodyCopy
      margin={0}
    >
      test lesson content
    </BodyCopy>
    <BodyCopy>
      <strong>
        Question : 
      </strong>
      test lesson question
    </BodyCopy>
  </styled.div>
</styled.div>
`;
